### PR TITLE
Preview most recent version of images

### PR DESCRIPTION
**Pull request by Galadirith**
This is a WIP towards #49

### CODE REVIEW

**Comment by Galadirith** [comment]
Still needs specs and currently doesn't condition on local or remote images (and I don't know how pathwatcher responds to remote images)

**Comment by leipert** [comment]
what is the refresh confidition?

File: lib/renderer.coffee [hunk 1]
```diff
@@ -97,9 +104,26 @@ resolveImagePaths = (html, filePath) ->
...
         unless fs.isFileSync(src)
-          img.attr('src', path.join(rootDirectory, src.substring(1)))
+          src = path.join(rootDirectory, src.substring(1))
+      else
+        src = path.resolve(path.dirname(filePath), src)
+
+      # Use most recent version of image
+      if imgVersion[src]?
+        if imgVersion[src] > 0
+          src = "#{src}##{imgVersion[src]}"
```
Comment (leipert) [inline lib/renderer.coffee]: I would like "#{src}?v=#{imgVersion[src]}" more :+1:

**Comment by Galadirith** [comment]
> what is the refresh condition? pathwatch tells you that the file changed?

Exactly that. Pathwatcher will watch for any changes on the file, and then increment the `imgVersion` array for `src` and then triggers all open previews to re-render.

The debounce is to cover multiple fires as described in atom/node-pathwatcher#50.

**Comment by Galadirith** [comment]
All specs are passing, and I think the specs I added for this PR give good coverage. I'll review tomorrow with fresher eye's and if everything's still looking good I'll merge in to master :D

**Comment by leipert** [comment]
Looks really nice, especially your test png :dancer: :+1:

**Comment by Galadirith** [comment]
@leipert Thanks for the eye's on :D

**Pull request merged**